koku/api/report/azure/openshift/provider_map.py

```diff
@@ -0,0 +1,553 @@
+#
+# Copyright 2019 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+"""Provider Mapper for OCP on Azure Reports."""
+
+from django.db.models import CharField, Count, DecimalField, F, Max, Sum, Value
+from django.db.models.functions import Coalesce
+
+from api.report.provider_map import ProviderMap
+from reporting.models import (
+    OCPAzureCostLineItemDailySummary,
+    OCPAzureCostLineItemProjectDailySummary,
+)
+
+
+class OCPAzureProviderMap(ProviderMap):
+    """OCP on Azure Provider Map."""
```

Comment:
Could I see more documentation for why this class exists?

Reply:
Hey @nbon12. It's a longer answer than it makes sense to type out here. Let's talk about it tomorrow in person.